news/tests/test_api.py

```diff
@@ -1,59 +1,142 @@
+from utils.setup_tests.comment import CommentFactory
+from utils.setup_tests.news import ApprovedNewsFactory, AuthoredNewsFactory, NewsFactory
+from utils.setup_tests.news_user import NewsUserFactory
+from utils.setup_tests.saved_news import SavedNewsFactory
+from utils.setup_tests.tags import TagsFactory
 
 
 class TestCategories(APITestCase):
     @classmethod
     def setUpTestData(cls):
         cls.categories = CategoriesFactory()
+        cls.news = NewsFactory()
 
     def test_get_list_categories(self):
         response = self.client.get("/api/Categories/", format="json")
         self.assertEqual(response.status_code, 200)
         self.assertEqual(self.categories.title, response.json()[0]["title"])
 
+    def test_success_get_news_by_categories(self):
+        title = self.news.title.split()[0]
+        response = self.client.get(f"/api/Categories/{title}/", format="json")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(self.news.title, response.json()[0]["title"])
+        self.assertEqual(self.news.description, response.json()[0]["description"])
+        self.assertEqual(self.news.content, response.json()[0]["content"])
+
+    def test_get_news_by_categories_not_found(self):
+        title = "rrrrrrrrrrrrrrrrrrr"
+        response = self.client.get(f"/api/Categories/{title}/", format="json")
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual("object not found", response.json())
+
 
 class TestTags(APITestCase):
     @classmethod
     def setUpTestData(cls):
         cls.tags = TagsFactory()
+        cls.news = NewsFactory()
 
     def test_get_list_tags(self):
         response = self.client.get("/api/Tags/", format="json")
         self.assertEqual(response.status_code, 200)
         self.assertEqual(self.tags.title, response.json()[0]["title"])
 
+    def test_success_get_news_by_title(self):
+        title = self.news.title.split()[0]
+        response = self.client.get(f"/api/Tags/{title}/", format="json")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(self.news.title, response.json()[0]["title"])
+        self.assertEqual(self.news.description, response.json()[0]["description"])
+        self.assertEqual(self.news.content, response.json()[0]["content"])
+
+    def test_get_news_by_title_not_found(self):
+        title = "rrrrrrrrrrrrrrrrrrr"
+        response = self.client.get(f"/api/Tags/{title}/", format="json")
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual("object not found", response.json())
+
 
 class TestAuthors(APITestCase):
     @classmethod
     def setUpTestData(cls):
         cls.author = AuthorFactory()
+        cls.news_by_author = AuthoredNewsFactory()
 
     def test_get_list_authors(self):
         response = self.client.get("/api/Authors/", format="json")
         self.assertEqual(response.status_code, 200)
+        self.assertEqual(self.author.name, response.json()[0]["name"])
```

Comment:
why you just check first author?Make smth like for index, author in enumerate(response)